Lib/platform.py

```diff
@@ -1230,6 +1230,68 @@ def platform(aliased=0, terse=0):
     _platform_cache[(aliased, terse)] = platform
     return platform
 
+### freedesktop.org os-release standard
+# https://www.freedesktop.org/software/systemd/man/os-release.html
+
+# NAME=value with optional quotes (' or ")
+_os_release_line = re.compile(
+    "^(?P<name>[a-zA-Z0-9_]+)=(?P<quote>[\"\']?)(?P<value>.*)(?P=quote)$"
+)
+# /etc takes precedence over /usr/lib
+_os_release_candidates = ("/etc/os-release", "/usr/lib/os-release")
+_os_release_cache = None
+
+
+def _parse_os_release(lines):
+    # These fields are mandatory fields with well-known defaults
+    # in pratice all Linux distributions override NAME, ID, and PRETTY_NAME.
+    info = {
+        "NAME": "Linux",
+        "ID": "linux",
+        "PRETTY_NAME": "Linux",
+    }
+
+    for line in lines:
+        mo = _os_release_line.match(line)
+        if mo is not None:
+            info[mo.group('name')] = mo.group('value')
+
+    # ID_LIKE is a space separated field of ids
+    if 'ID_LIKE' in info:
+        info['ID_LIKE'] = tuple(
+            id_like for id_like in info['ID_LIKE'].split(' ') if id_like
+        )
+
+    return info
+
+
+def freedesktop_os_release(candidates=_os_release_candidates):
+    """Return operation system identification from freedesktop.org os-release
+    """
+    global _os_release_cache
+
+    errno = None
+
+    if _os_release_cache is None:
+        for candidate in candidates:
+            try:
+                with open(candidate, encoding="utf-8") as f:
+                    _os_release_cache = _parse_os_release(f)
+                break
+            except OSError as e:
+                errno = e.errno
```

Comment:
Rather than storing the exception or its errno attribute, maybe use enumerate(candidates) and raise if it's the last candidate. Something like:

```
for index, candiate in enumerate(candidates): ...
```

except block:

```
if index == len(candidates)-1: raise
# else: ignore and try next candiate
```

Reply:
No, that would be confusing. The default file is the first file in the list. All except Alpine either have ``/etc/os-release`` or a symlink ``/etc/os-release -> /usr/lib/os-release``. I want the error message to contain the common case.